core/src/smartcontracts/isi/permissions/has_token.rs

```diff
@@ -42,6 +45,16 @@ pub struct HasTokenAsValidator<H: HasToken> {
     has_token: H,
 }
 
+impl<H: HasToken> Display for HasTokenAsValidator<H> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(
+            f,
+            "Allow if signer has the corresponding `{}` permission token",
```

Comment:
```suggestion
            "Allow if signer has the `{}` permission token",
```

Reply:
I'd like to keep `corresponging` word here, because *permission token* name does not fully identify it. *Signer* can have a permission token with the same name but with different parameters and that won't work

Reply:
That's a good point. Should we allow permission tokens with the same name?

Reply:
That is also related to #2517. I think duplicating names is okay. It's like asset definition id. We can replace **name** with `PermissionTokenDefinitionId` someday